std/datetime.d

```diff
@@ -29929,7 +29929,7 @@ else version(Windows)
         }
 
 
-        this(string name, TIME_ZONE_INFORMATION tzInfo) @safe immutable pure
+        this(string name, TIME_ZONE_INFORMATION tzInfo) @trusted immutable pure
```

Comment:
I'm more than a little unsure about this. The conversion on line 29934 is definitely wrong as is (conversion from wchar* to string does not work, and fails in ugly ways at runtime). The question is, should this function be @trusted? @system?
Should it even be using this conversion? Am I right to address the issue here or should that be a separate pull?

Regarding to the safety of using to!string(wchar*), MSDN claims in one place that DaylightName will always be zero-terminated. If this is true, this should be safe. However, there are many versions of the info page on TIME_ZONE_INFORMATION, and not all of them include this information. The constructor is private though, and I haven't found a case where it could be called with invalid data.